WordPress/src/main/java/org/wordpress/android/util/WPUrlUtils.java

```diff
@@ -0,0 +1,45 @@
+package org.wordpress.android.util;
+
+import java.net.URI;
+import java.net.URL;
+
+public class WPUrlUtils {
```

Comment:
Methods of this class are called in a lot of places in the app. They're also used in the XML-RPC client, and in Async Tasks. Thus they need to be `synchronized`.
EX: `RefreshBlogContentTask` is running in background, and the user switches to the Reader, or to Notifications. Both the XML-RPC client in RefreshBlogContentTask, and the Reader or Notifications can call one of those methods above. If the 2 calls arrive at almost the same time, we can get back unexpected results.